src/agentscope/agents/react_agent.py

```diff
@@ -129,13 +94,22 @@ def __init__(
         self.memory.add(Msg("system", self.sys_prompt, role="system"))
 
         # Initialize a parser object to formulate the response from the model
-        self.parser = MarkdownJsonDictParser(
-            content_hint={
-                "thought": "what you thought",
-                "speak": "what you speak",
-                "function": service_toolkit.tools_calling_format,
-            },
-            required_keys=["thought", "speak", "function"],
+        self.parser = RegexTaggedContentParser(
+            format_instruction="""Respond with specific tags as outlined below:
+
+- When not calling any tool functions:
+<thought>what you thought</thought>
+<speak>what you speak</speak>
+
+- When calling tool functions, note the "arg_name" should be replaced with the actual argument name:
+<thought>what you thought</thought>
+<speak>what you speak</speak>
```

Comment:
I am not sure whether the <speak> here is a must for ReAct agent. Other frameworks found that <thought> + <action> performs the best. And in practice, both <thought> and <action> should be spoken by the agent to let the user know the internal process of agent anyway.

Reply:
Improved by providing a response field in the last.